classes/UpgradeTools/CoreUpgrader/CoreUpgrader80.php

```diff
@@ -0,0 +1,128 @@
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\AutoUpgrade\UpgradeTools\CoreUpgrader;
+
+use PrestaShop\Module\AutoUpgrade\UpgradeException;
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\MailTemplate\Command\GenerateThemeMailTemplatesCommand;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+
+class CoreUpgrader80 extends CoreUpgrader
+{
+    protected function initConstants()
+    {
+        parent::initConstants();
+
+        // Container may be needed to run upgrade scripts
+        $this->container->getSymfonyAdapter()->initAppKernel();
+    }
+
+    protected function upgradeDb($oldversion)
+    {
+        parent::upgradeDb($oldversion);
+
+        $commandResult = $this->container->getSymfonyAdapter()->runSchemaUpgradeCommand();
+        if (0 !== $commandResult['exitCode']) {
+            throw (new UpgradeException($this->container->getTranslator()->trans('Error upgrading Doctrine schema', array(), 'Modules.Autoupgrade.Admin')))
+                ->setQuickInfos(explode("\n", $commandResult['output']));
+        }
+    }
+
+    protected function upgradeLanguage($lang)
+    {
+        $isoCode = $lang['iso_code'];
+
+        if (!\Validate::isLangIsoCode($isoCode)) {
+            return;
+        }
+        $errorsLanguage = array();
+
+        if (!\Language::downloadLanguagePack($isoCode, _PS_VERSION_, $errorsLanguage)) {
+            throw new UpgradeException(
+                $this->container->getTranslator()->trans(
+                    'Download of the language pack %lang% failed. %details%',
+                    [
+                        '%lang%' => $isoCode,
+                        '%details%' => implode('; ', $errorsLanguage),
+                    ],
+                    'Modules.Autoupgrade.Admin'
+                )
+            );
+        }
+
+        $lang_pack = \Language::getLangDetails($isoCode);
+        \Language::installSfLanguagePack($lang_pack['locale'], $errorsLanguage);
+
+        if (!$this->container->getUpgradeConfiguration()->shouldKeepMails()) {
+            $mailTheme = \Configuration::get('PS_MAIL_THEME', null, null, null, 'modern');
+
+            $frontTheme = _THEME_NAME_;
+            $frontThemeMailsFolder = _PS_ALL_THEMES_DIR_ . $frontTheme . '/mails';
+            $frontThemeModulesFolder = _PS_ALL_THEMES_DIR_ . $frontTheme . '/modules';
+
+            $generateCommand = new GenerateThemeMailTemplatesCommand(
+                $mailTheme,
+                $lang_pack['locale'],
+                true,
+                $frontThemeMailsFolder,
+                $frontThemeModulesFolder
+            );
+            /** @var CommandBusInterface $commandBus */
+            $commandBus = $this->container->get('prestashop.core.command_bus');
```

Comment:
It seems that this `container` is not what you think 😅

Reply:
t's not? What is it then? It's not the Symfony container?

Reply:
@jolelievre nope ^^' it's this : https://github.com/PrestaShop/autoupgrade/blob/dev/classes/UpgradeContainer.php

Reply:
Damn 😅. I admit I didn't check if it worked at all... shame!

Does it mean there is no way to use the Symfony container or any services it might contain? Should I reimplement the whole thing? (seems unrealistic)
Or can I get the container with `SymfonyContainer::getInstance` for example?

Reply:
@jolelievre you should be able to do something like this in: https://github.com/PrestaShop/autoupgrade/blob/dev/classes/UpgradeTools/ModuleAdapter.php
```php
public function getCommandBus()
{
    if (null === $this->commandBus) {
        $this->commandBus = $this->symfonyAdapter
            ->initAppKernel()
            ->getContainer()
            ->get('prestashop.core.command_bus');
    }

    return $this->commandBus;
}
```
and then use `$commandBus = $this->container->getModuleAdapter()->getCommandBus();`